Gems/PhysX/Code/Source/PrismaticJointComponent.cpp

```diff
@@ -84,4 +86,104 @@ namespace PhysX
             m_jointSceneOwner = leadFollowerInfo.m_followerBody->m_sceneOwner;
         }
     }
+
+    void PrismaticJointComponent::Activate()
+    {
+        const AZ::EntityComponentIdPair id(GetEntityId(), GetId());
+        PhysX::JointInterfaceRequestBus::Handler::BusConnect(id);
+        JointComponent::Activate();
+    };
+
+    void PrismaticJointComponent::Deactivate()
+    {
+        PhysX::JointInterfaceRequestBus::Handler::BusDisconnect();
+        JointComponent::Deactivate();
+    };
+
+    physx::PxD6Joint* PrismaticJointComponent::GetPhysXD6Joint() const
+    {
+        if (m_native)
+        {
+            return m_native;
+        }
+        auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        AZ_Assert(sceneInterface, "No sceneInterface");
+        const auto* joint = sceneInterface->GetJointFromHandle(m_jointSceneOwner, m_jointHandle);
+        physx::PxJoint* native = static_cast<physx::PxJoint*>(joint->GetNativePointer());
+        if (native && native->is<physx::PxD6Joint>())
+        {
+            physx::PxD6Joint* nativeD6 = static_cast<physx::PxD6Joint*>(joint->GetNativePointer());
+            m_native = nativeD6;
+            return nativeD6;
+        }
+        return nullptr;
+    }
+
+    float PrismaticJointComponent::GetPosition() const
+    {
+        auto* native = GetPhysXD6Joint();
+        if (native)
+        {
+            // Underlying PhysX joint is D6, but it simulates PhysXPrismatic joint.
+            // The D6 joint has only X-axis unlocked, so report only X travel.
+            return native->getRelativeTransform().p.x;
+        }
+        return 0.f;
+    };
+
+    float PrismaticJointComponent::GetVelocity() const
+    {
+        auto* native = GetPhysXD6Joint();
+        if (native)
+        {
+            // Undelying PhysX joint is D6, but it simulates PhysXPrismatic joint.
+            // The D6 joint has only X-axis unlocked, so report only X velocity.
+            return native->getRelativeLinearVelocity().x;
+        }
+        return 0.f;
+    };
+
+    AZStd::pair<float, float> PrismaticJointComponent::GetLimits() const
+    {
+        auto* native = GetPhysXD6Joint();
+        if (native)
+        {
+            auto limits = native->getLinearLimit(physx::PxD6Axis::eX);
+            return AZStd::pair<float, float>(limits.lower, limits.upper);
+        }
+        return AZStd::make_pair(0.f, 0.f);
+    }
+
+    AZ::Transform PrismaticJointComponent::GetTransform() const
+    {
+        auto* native = GetPhysXD6Joint();
+        if (native)
+        {
+            const auto worldFromLocal = native->getRelativeTransform();
+            return AZ::Transform(
```

Comment:
See if `return PxMathConvert(worldFromLocal);` works here.
There are a bunch of utility conversion functions in Gems\PhysX\Code\Include\PhysX\MathConversion.h